src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -353,45 +360,107 @@ private void barrierOnRebalanceReached(final Partition.PartitionKey pk) {
     }
 
     private void reconfigureKafkaConsumer(final boolean forceSeek) {
-        if (kafkaConsumer == null) {
+        if (eventConsumer == null) {
             throw new IllegalStateException(
                     "kafkaConsumer should not be null when calling reconfigureKafkaConsumer method");
         }
-        final Set<Partition.PartitionKey> currentKafkaAssignment = kafkaConsumer.assignment().stream()
-                .map(tp -> new Partition.PartitionKey(tp.topic(), String.valueOf(tp.partition())))
-                .collect(Collectors.toSet());
-
-        final Set<Partition.PartitionKey> currentNakadiAssignment = offsets.keySet().stream()
+        final Set<TopicPartition> newAssignment = offsets.keySet().stream()
                 .filter(o -> !this.releasingPartitions.containsKey(o))
                 .collect(Collectors.toSet());
-        if (!currentKafkaAssignment.equals(currentNakadiAssignment) || forceSeek) {
-            final Map<Partition.PartitionKey, TopicPartition> kafkaKeys = currentNakadiAssignment.stream().collect(
-                    Collectors.toMap(
-                            k -> k,
-                            k -> new TopicPartition(k.getTopic(), Integer.valueOf(k.getPartition()))));
-            // Ignore order
-            kafkaConsumer.assign(new ArrayList<>(kafkaKeys.values()));
-            // Check if offsets are available in kafka
-            kafkaConsumer.seekToBeginning(kafkaKeys.values().toArray(new TopicPartition[kafkaKeys.size()]));
-            kafkaKeys.forEach((key, kafka) -> offsets.get(key).ensureDataAvailable(kafkaConsumer.position(kafka)));
-            //
-            kafkaKeys.forEach((k, v) -> kafkaConsumer.seek(v, offsets.get(k).getSentOffset() + 1));
-            offsets.values().forEach(PartitionData::clearEvents);
+        if (forceSeek) {
+            // Force seek means that user committed somewhere not within [commitOffset, sentOffset], and one
+            // should fully reconsume all the information from underlying storage. In order to do that, we are
+            // removing all the current assignments for real consumer.
+            try {
+                eventConsumer.reassign(Collections.emptyList());
+            } catch (final NakadiException | InvalidCursorException ex) {
+                throw new NakadiRuntimeException(ex);
+            }
         }
+        final Set<TopicPartition> currentKafkaAssignment = eventConsumer.getAssignment().stream()
```

Comment:
I expected not to find `Kafka` word in this class. This is about the assignment of the underlying event consumer.